Riccati Equation Solver

romgerale opened the pull request:
Riccati Equation Solver, a proposal for the addition of an algebraic Riccati Equation Solver in the package org.hipparchus.linear.
It also incluses extensions in the Array2DRowRealMatrix to support Kronecker products and ComplexEigenDecomposition.

romgerale commented:
Done.

maisonobe commented:
I have merge the changes into the master branch.
Please take a look at what I have done to check I did do any mistake. I am not sure about the javadoc
parameter description I used.
In order to meet our development conventions, I have:

  - removed trailing blanks
  - removed tabs
  - replaced hard coded error messages with proper exceptions specifiers
  - completed all javadoc
  - fixed checkstyle errors
  - fixed findbugs errors

I am not sure the stack and unstack operations should really be in the Array2DRowRealMatrix class. They seem specific, are used only in the RiccatiEquationSolverImpl, and don't really need access
to private fields. Perhaps they should be private methods within RiccatiEquationSolverImpl.

romgerale commented:
Dear all,
         I checked the javadoc and they are OK.
         Regarding stack,unstack,kroneckerProduct methods, the idea was that they are well-defined operations on RealMatrixes like add, subtract and multiply, nonetheless, the change in the interface RealMatrix was not proposed in order to avoid the implementation in all the concrete classes of it. Do you agree?
         A small proposal in the Hipparchus-geometry, it would be great if a Vector3D has an operation to return the skew-symmetric tensor (such RealMatrix is very common in the dynamics). What do you think?
Best Regards,Alessandro Romero.

      From: maisonobe <notifications@github.com>
 To: Hipparchus-Math/hipparchus <hipparchus@noreply.github.com> 
Cc: romgerale <romgerale@yahoo.com.br>; Author <author@noreply.github.com>
 Sent: Monday, November 13, 2017 12:25 PM
 Subject: Re: [Hipparchus-Math/hipparchus] Riccati Equation Solver (#29)
   
I have merge the changes into the master branch.
Please take a look at what I have done to check I did do any mistake. I am not sure about the javadoc
parameter description I used.
In order to meet our development conventions, I have:   
   - removed trailing blanks
   - removed tabs
   - replaced hard coded error messages with proper exceptions specifiers
   - completed all javadoc
   - fixed checkstyle errors
   - fixed findbugs errors
I am not sure the stack and unstack operations should really be in the Array2DRowRealMatrix class. They seem specific, are used only in the RiccatiEquationSolverImpl, and don't really need access
to private fields. Perhaps they should be private methods within RiccatiEquationSolverImpl.—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
